### DISCUSSION TITLE
Enlarging NavigationView's TogglePaneButton

### DISCUSSION THREAD

**Discussion by JGKle** [amwx/FluentAvalonia#599] [Q&A]
Forgive what I'm sure is a dumb question - I'm still new to styling in Avalonia.

I can't for the life of me figure out how to enlarge the NavigationView's TogglePaneButton. When I use the inspector, setting the height/width/fontsize yields no change; likewise if I drill all the way down to the TextBlock that shows the icon & try to set the font size there.

I did find that if I enlarge the Border directly inside the TogglePaneButton, the clickable area does increase (but not the icon of course). However, even that I'm having trouble doing with styles:

```
<Grid.Styles>
  <Style Selector="Button.TogglePaneButton Border">
    <Setter Property="Width" Value="50"/>
    <Setter Property="Height" Value="50"/>
  </Style>
</Grid.Styles>
```

The default size feels pretty small & hard to tap on mobile screens. I'm sure I'm missing an easy way to do this...any help would be appreciated

**Comment by JGKle** [comment]
....Anyone? 🙏

**Reply by timunie** [reply]
Here is the original ControlTheme: https://github.com/amwx/FluentAvalonia/blob/master/src/FluentAvalonia/Styling/ControlThemes/FAControls/NavigationView/NavigationViewBackButtonStyles.axaml 

You can either override these Resources:https://github.com/amwx/FluentAvalonia/blob/2264dfe44bce8b0f532f58909a47cd5edf9acdf7/src/FluentAvalonia/Styling/ControlThemes/FAControls/NavigationView/NavigationViewBackButtonStyles.axaml#L16-L17

or create a new ControlTheme with the given x:Key: 

```xml
<ControlTheme x:Key="NavigationBackButtonNormalStyle" BasedOn="{StaticResouce NavigationBackButtonNormalStyle}" TargetType="Button">
   <Setter Property="Height" Value="100" />
   [...]
```

**Reply by JGKle** [reply]
NavigationBackButtonWidth doesn't seem to be it. To clarify, I'm referring to this:

![2024-11-02 20 24 33](https://github.com/user-attachments/assets/a0669b53-7308-43d7-a469-d76dc9e4c5f0)

Even if I directly edit `FluentAvalonia/Styling/ControlThemes/FAControls/NavigationView/NavigationViewBackButtonStyles.axaml` in the sample app & edit `NavigationBackButtonWidth`, the size of that button doesn't change. Using the inspector, I think it's the `TogglePaneButton`, per the OP. But no matter what I tweak I can't get the button indicated above to increase in size.

**Reply by JGKle** [reply]
...Anyone: 🙏

**Reply by timunie** [reply]
please use DevTools via F12 and explore the given ControlTheme I linked above.

**Reply by JGKle** [reply]
I did - that's what I meant by `Using the inspector, I think it's the TogglePaneButton, per the OP. ` That's how I found TogglePaneButton. Whatever I tried I couldn't figure out anything that worked.

**Reply by JGKle** [reply]
For what it's worth, I also spent 3 hours with various agentic coding tools, given them full access to the FluentAvalonia source, and none were able to resize the button (tried Claude Code, OpenHands, and Aider). That mirrored my experience.

I'm trying to get the button to fill the top bar, like this:

![2025-04-27 15 23 07](https://github.com/user-attachments/assets/d7a7a33e-7005-466d-ba49-26a4288f24ec)

whereas by default it's like this:

![2025-04-27 15 23 35](https://github.com/user-attachments/assets/cbe83c7b-84b9-411b-afbc-f02c1a9c9811)

Still stuck to no avail. I just don't see how it can be done with the response you gave above :(
